packages/core/admin/admin/src/translations/en.json

```diff
@@ -186,31 +186,15 @@
-  "Settings.permissions.auditLogs.entry.publish": "Publish entry {model, select, undefined {} other {({model})}}",
-  "Settings.permissions.auditLogs.entry.unpublish": "Unpublish entry {model, select, undefined {} other {({model})}}",
-  "Settings.permissions.auditLogs.media.create": "Create media",
-  "Settings.permissions.auditLogs.media.update": "Update media",
-  "Settings.permissions.auditLogs.media.delete": "Delete media",
-  "Settings.permissions.auditLogs.user.create": "Create user",
-  "Settings.permissions.auditLogs.user.update": "Update user",
-  "Settings.permissions.auditLogs.user.delete": "Delete user",
-  "Settings.permissions.auditLogs.admin.auth.success": "Admin login",
-  "Settings.permissions.auditLogs.admin.logout": "Admin logout",
-  "Settings.permissions.auditLogs.content-type.create": "Create content type",
-  "Settings.permissions.auditLogs.content-type.update": "Update content type",
-  "Settings.permissions.auditLogs.content-type.delete": "Delete content type",
-  "Settings.permissions.auditLogs.component.create": "Create component",
-  "Settings.permissions.auditLogs.component.update": "Update component",
-  "Settings.permissions.auditLogs.component.delete": "Delete component",
-  "Settings.permissions.auditLogs.role.create": "Create role",
-  "Settings.permissions.auditLogs.role.update": "Update role",
-  "Settings.permissions.auditLogs.role.delete": "Delete role",
-  "Settings.permissions.auditLogs.permission.create": "Create permission",
-  "Settings.permissions.auditLogs.permission.update": "Update permission",
-  "Settings.permissions.auditLogs.permission.delete": "Delete permission",
+  "Settings.permissions.auditLogs.actions.admin": "Admin {action, select, authsuccess {login} logout {logout} other {}}",
+  "Settings.permissions.auditLogs.actions.component": "{action, select, create {Create} update {Update} delete {Delete} other {}} component",
+  "Settings.permissions.auditLogs.actions.content-type": "{action, select, create {Create} update {Update} delete {Delete} other {}} content type",
+  "Settings.permissions.auditLogs.actions.entry": "{action, select, create {Create} update {Update} delete {Delete} publish {Publish} unpublish {Unpublish} other {}} entry {model, select, undefined {} other {({model})}}",
+  "Settings.permissions.auditLogs.actions.media": "{action, select, create {Create} update {Update} delete {Delete} other {}} media",
+  "Settings.permissions.auditLogs.actions.media-folder": "{action, select, create {Create} update {Update} delete {Delete} other {}} media folder",
+  "Settings.permissions.auditLogs.actions.permission": "{action, select, create {Create} update {Update} delete {Delete} other {}} permission",
+  "Settings.permissions.auditLogs.actions.role": "{action, select, create {Create} update {Update} delete {Delete} other {}} role",
+  "Settings.permissions.auditLogs.actions.user": "{action, select, create {Create} update {Update} delete {Delete} other {}} user",
```

Comment:
What if we take this to its extreme?

```suggestion
  "Settings.permissions.auditLogs.action": "{actionName, select, admin {Admin} other {}} {action, select, authsuccess {login} logout {logout} create {create} update {update} delete {delete} publish {publish} unpublish {unpublish} other {}} {type, select, media {media} folder {media folder} permission {permission} role {role} user {user} model {({model})} other {} }",
```

What I like about this:

- it is very concise and it allows to keep things together semantically
- it makes it very explicit what the values can be; with dynamic IDs it is very hard (without TS) to understand what `actionName` can become
- I agree the syntax is not pretty, but that's what we have and I think we should utilize it to the max. We could make this a bigger topic and say: the way we've implemented i18n is completely wrong. Hardcoded IDs should be discouraged and usually you wouldn't maintain these JSON files by hand and use extractors instead. Once you've set it up that way you can use string literals when calling `formatMessage` and properly format them multiline: https://formatjs.io/docs/getting-started/message-extraction They way we use dynamic IDs make it impossible to properly set it up one day
- I find it very flexible: instead of falling back to `{}` in the other branches you could also just print the value that `select` is being run on. This allows to extend these strings and users have chance to simplify add another case to the select in case something is not translated.

Reply:
> usually you wouldn't maintain these JSON files by hand and use extractors instead
Love that idea, this: https://formatjs.io/docs/getting-started/message-extraction/ is very cool, I didn't know that was a thing

> I find it very flexible: instead of falling back to {} in the other branches you could also just print the value that select is being run on.

Just to be clear you mean something like this: `{arg, select, option {Something} other {{arg}}}`